pyuvdata/cst_beam.py

```diff
@@ -177,6 +176,11 @@ def read_cst_beam(self, filename, beam_type='power', feed_pol='x',
         self.axis2_array = theta_axis
         self.Naxes2 = self.axis2_array.size
 
+        if np.any(np.core.defchararray.find(np.core.defchararray.lower(np.array(column_names)), 'abs(dir.)') != -1):
+            self.data_normalization = 'solid_angle'
+        elif np.any(np.core.defchararray.find(np.core.defchararray.lower(np.array(column_names)), 'abs(v)') != -1):
+            self.data_normalization = 'physical'
+
```

Comment:
It appears to me that you're doing this search twice, here and below where you define `data_col`. I think you should do it once to figure out whether it's a directivity or not and then use the appropriate string when you define `data_col`. That limits the code replication and ensures that you can't get a weird combination (mixed column types).

Reply:
Actually, I'm confused by something else here. It looks like you're using the power column names to decide on the normalization whether or not you're reading in a power or an efield beam. Is that the right behavior? Or should the units (dbi) be used as an indicator? What if the power wasn't written to the file along with the efield?

Reply:
You're right. I think I accidentally assumed that dBi simply stood for dB but it actually stands for decibal Isotropic which is unambiguously referring to solid angle normalization. I should therefor eliminate this search and normalize the solid angle if units are in "dBi" and physical if units are in "V". I've removed this extraneous search and changed the normalization setting to be dependent on "mag_dbi". I've also added a check that the units are in dBi or V and raise a type error if unexpected units come up so we can explicitly add them as needed.